test/end2end_test.go

```diff
@@ -7433,29 +7445,30 @@ func (s *httpServer) start(t *testing.T, lis net.Listener) {
 	}()
 }
 
-func doHTTPHeaderTest(t *testing.T, errCode codes.Code, headerFields ...[]string) {
+func doHTTPHeaderTest(t *testing.T, errCode codes.Code, headerFields ...[]string) error {
 	lis, err := net.Listen("tcp", "localhost:0")
 	if err != nil {
-		t.Fatalf("Failed to listen. Err: %v", err)
+		return fmt.Errorf("Failed to listen. Err: %v", err)
```

Comment:
Nit: errors returned from this function could be rephrased to align more with go style.
fmt.Errorf("net.Listen(tcp, localhost:0) failed: %v", err)
fmt.Errorf("grpc.Dial(%v, grpc.WithInsecure()) failed: %v", lis.Addr().String(), err)
fmt.Errorf("client.FullDuplexCall(ctx) failed: %v", err)
Thanks.

Reply:
The changes here are reverted, so I'm not changing these lines now.
There are more error messages like this in our test, we should probably do a cleanup someday...

Reply:
Ack.